src/plugins/intel_gpu/src/graph/impls/cpu/non_max_suppression.cpp

```diff
@@ -39,18 +38,21 @@ std::vector<result_indices> run_nms(
     float score_threshold,
     float iou_threshold,
     float soft_nms_sigma,
-    bool sort_result_descending
+    bool sort_result_descending,
+    bool soft_nms
 ) {
     auto less = [](const boxInfo& l, const boxInfo& r) {
         return l.score < r.score || ((l.score == r.score) && (l.idx > r.idx));
     };
     float scale = 0.0f;
     if (soft_nms_sigma > 0.0f) {
         scale = -0.5f / soft_nms_sigma;
+    } else {
+        soft_nms = false;
     }
     auto coeff = [&](float iou) {
         const float weight = std::exp(scale * iou * iou);
-        return iou <= iou_threshold ? weight : 0.0f;
+        return iou <= iou_threshold || soft_nms ? weight : 0.0f;
```

Comment:
just for readability may it worth to take conditions in braces, e.g. 
return( iou <= iou_threshold || soft_nms) ? weight : 0.0f;

Reply:
done